File: src/main/java/com/janet/Hello.java
package com.janet;

public class Hello {
    public static void main(String[] args) {
       // System.out.println("Hello world!");
       // new Person().hello();
        String s = "abcde";
        System.out.println(s.charAt(3));
        System.out.println(s.substring(1,4));
        Person p = new Person( "janet",66.5f,17.2f);
        p.hello();
        p.weight = 66.5f;
        p.height = 17.2f;
        p.bmi();
        System.out.println(p.bmi());
        int score = 88;
        System.out.println(score < 80 || score > 90);
        char c = 'A';
        System.out.println(c < 'a');
       /* int age = 19;
        Integer age2 = 19;
        char c = 'm';
        byte b = 120;
        float weight = 55.5f;
        double height = 17.2; */
      }

}

File: src/main/kotlin/com/janet/hello.kt
package com.janet

fun main(args: Array<String>) {
 //println("Hello kotlin")
  //  Human().hello()
    var s : String? = "abcde" //?讓s可以為 null
    s = null //不會出現 nullpointerexception
    println(s?.length)  //* s!! null exception 要被處理，不當調
    println(s?.get(3))
    println(s?.substring(3))
    val h = Human(weight = 55.5f, height = 17.2f)
    h.hello()
    println(h.bmi())
    val score = 88;
    println(score > 80)
    val c : Char = 'A'
    println(c.toInt())
    println(c.toInt() > 60)
   /* val age : Int = 19 // value 不能改變
    val age2 = 19
    var weight = 55.5f // variable 未來可改變
    var name : String
    name = "janet"*/
}
class Human(var name: String="", var weight: Float, var  height: Float){
    init {
        println("test $weight")
    }
    //constructor(name:String,weight: Float,height: Float):this(weight,height)
    fun bmi() : Float {
        val bmi = weight / (height*height)
        return  bmi
    }
    fun hello() {
        println("Hello kotlin")
    }
}


